### ISSUE TITLE
Detection rule id 'jwt' cannot detect jwt leaks

### ISSUE THREAD

**Issue by xiaotaozi1127** [gitleaks/gitleaks#1208]
**Describe the bug**
when using default gitleaks configuration, rule id 'jwt' cannot detect a normal json web token. However, when i create a customized `.gitleaks.toml `under the root of repo and change the rule id from 'jwt' to 'my-jwt', it works again.

**To Reproduce**
i test with this jwt `val test = eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c`, when i try to commit code with this value, i can commit it and there is no leaks warning from pre-commit hook. 

However, if i create `.gitleaks.toml` and copy the gitleaks jwt rule and replace the rule id from `jwt` to `my-jwt` or any other value, it can detect the jwt leaks. Not sure why the name of rule id matters? 

`
[[rules]]
description = "JSON Web Token"
id = "my jwt"
regex = '''(?i)\b(ey[0-9a-z]{10,34}\.ey[0-9a-z-\/_]{30,500}\.[0-9a-zA-Z-\/_]{10,200}={0,2})(?:['|\"|\n|\r|\s|\x60|;]|$)'''
keywords = [
    "ey",
]
`

<img width="748" alt="image" src="https://github.com/gitleaks/gitleaks/assets/4277354/a56bfa2d-a068-40ec-b71e-4ef42ee02fe3">

**Expected behavior**
I expect default jwt rule can detect my jwt leaks.

**Basic Info (please complete the following information):**
 - OS: MAC OS
 - Gitleaks Version: v8.17.0

**Additional context**

cc @zricethezav

**Comment by rgmz** [comment]
Can you share the full command and output? I quickly tested this with the latest version and can't seem to reproduce it.

```sh
$ ./gitleaks detect --verbose --source /tmp/gitleaks-test/

    ○
    │╲
    │ ○
    ○ ░
    ░    gitleaks

Finding:     val test = eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE...c
Secret:      eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE...
RuleID:      jwt
Entropy:     5.444070
File:        mystuff.txt
Line:        1
Commit:      5e00309e9a6f4f2a4b10eddb970cd3ea2f212a0d
Author:      Richard Gomez
Email:       richard@example.com
Date:        2023-06-23T14:04:37Z
Fingerprint: 5e00309e9a6f4f2a4b10eddb970cd3ea2f212a0d:mystuff.txt:jwt:1

10:04AM INF 1 commits scanned.
10:04AM INF scan completed in 64.5ms
10:04AM WRN leaks found: 1
```

**Comment by rgmz** [comment]
I did notice that `gitleaks protect` seems to have a blindspot for new files, however, I don't think this is related to your problem.

```sh
# New file that is untracked.
 $ git status
On branch master
Untracked files:
  (use "git add <file>..." to include in what will be committed)
        BA.kt

nothing added to commit but untracked files present (use "git add" to track)

# Gitleaks protect does not pick it up.
$ gitleaks protect --verbose --config custom.toml --source /tmp/gitleaks-test/

10:10AM INF 0 commits scanned.
10:10AM INF scan completed in 57.7ms
10:10AM INF no leaks found

# Track the new file
$ git add BA.kt
$ git status
On branch master
Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
        new file:   BA.kt

# Scan again, still nothing.
$ gitleaks protect --verbose --config custom.toml --source /tmp/gitleaks-test/

10:10AM INF 0 commits scanned.
10:10AM INF scan completed in 57.7ms
10:10AM INF no leaks found
```

(Technically this is expected behavior with `git diff`, but it does seem like a rough edge @zricethezav.)

**Comment by xiaotaozi1127** [comment]
Hi, thanks for your reply. I found the cause now.

At the beginning, i created a .gitleaksignore file and ignored this line with this fingerprint:
`
app/src/main/kotlin/deliveryteaminkotlin/BA.kt:jwt:7
`

Then, I deleted the fingerprint locally, but I didn't commit the changes, so gitleaks will think we still have such ignore configuration. It can no longer report this jwt leak unless we change the rule ID.

After I commit the .gitleaksignore changes, it can successfully detect it.

Sorry I submitted an error report, it can be closed
